src/exanb/grid_cell_particles/include/exanb/grid_cell_particles/generate_particle_lattice.h

```diff
@@ -0,0 +1,579 @@
+      n_particles_cell = 4;
+      if( types.size() != n_particles_cell )
+        {
+          lerr << "Parameter types for double body centered tetragonal lattice must contain exactly 4 names" << std::endl;
+          std::abort();
+        }
+      positions.resize(n_particles_cell);
+      positions = {
+        {.00, .00, .00} ,
+        {.00, .50, .25} ,
+        {.50, .50, .50} ,
+        {.50, .00, .75} };	
+    }           
+
+    ParticleTypeMap typeMap;
+    if( ! particle_type_map.empty() )
+    {
+      typeMap = particle_type_map;
+    }
+    else
+    {
+      for(const auto & type_name : types)
+      {
+        const auto type_id = typeMap.size();
+        typeMap[ type_name ] = type_id;
+      }
+    }
+
+    // initialization of localization based particle filter (regions and masking)
+    PartcileLocalizedFilter<GridT,LinearXForm> particle_filter = { grid, { domain.xform() } };
+    particle_filter.initialize_from_optional_parameters( particle_regions,
+                                                         region,
+                                                         grid_cell_values, 
+                                                         grid_cell_mask_name, 
+                                                         grid_cell_mask_value,
+                                                         user_function,
+                                                         user_threshold );
+
+    // Get max and min positions
+    // Need to define the size of the box
+    // NOTE : only one processor need to do that   
+       
+    const IJK local_grid_dim = grid.dimension();
+    unsigned long long next_id = 0;      
+    
+    if( has_field_id )
+    {
+      size_t n_cells = grid.number_of_cells();
+      auto cells = grid.cells();
+#       pragma omp parallel for schedule(dynamic) reduction(max:next_id)
+      for(size_t cell_i=0;cell_i<n_cells;cell_i++) if( ! grid.is_ghost_cell(cell_i) )
+      {
+        size_t n_particles = cells[cell_i].size();
+        for(size_t p=0;p<n_particles;p++)
+        {
+          const unsigned long long id = cells[cell_i][field::id][p];
+          next_id = std::max( next_id , id );
+        }
+      }
+      MPI_Allreduce(MPI_IN_PLACE,&next_id,1,MPI_UNSIGNED_LONG_LONG,MPI_MAX,comm);
+      ++ next_id; // start right after gretest id
+    }
+
+    // =============== Section that concerns the porosity mode ========================
+
+    std::random_device rd{};
+    std::mt19937 gen{rd()};
+    std::default_random_engine generator;
+    std::normal_distribution<double> distribution_radius(void_mean_diameter, void_mean_diameter/2.);
+    std::uniform_real_distribution<double> distribution_center_x( domain.origin().x , domain.extent().x );
+    std::uniform_real_distribution<double> distribution_center_y( domain.origin().y , domain.extent().y );
+    std::uniform_real_distribution<double> distribution_center_z( domain.origin().z , domain.extent().z );      
+    
```

Comment:
Another interesting field to noise could be velocities. However, other operators can do it too.